### PR TITLE
Update JSCH and Conscrypt

### LINKED ISSUE #600: Can't clone from GitHub
**Describe the bug**

I cannot clone neither a private or public repository from GitHub.

**To Reproduce**

1. Go to Settings -> SSH keys -> Generate key (RSA) -> copy the public part to GitHub account (https://github.com/settings/keys)
2. On main screen click +
3. Provide Remote URL: `git@github.com:m-radzikowski/myprivaterepo.git`
4. Click Clone
5. Error window: Invalid remove, Clone failed

**Expected behavior**

Repo should be cloned.

**Smartphone (please complete the following information):**
 - Device: Redmi Note 10
 - OS: Android 11
 - App Version: 1.5.10

**Additional context**

I cannot clone any repo - I also tried this repo with URL: `git@github.com:maks/mgit.git`, got the same error.

The error does not tell much about what is actually wrong.

I have MFA enabled on the github account, but this shouldn't matter if I use SSH URL, right?

### LINKED ISSUE #601: RSA key with SHA-1 no longer alllowed.
git@github.com:username/repo_name.git: 

*ERROR*: You're using an RSA key with SHA-1 which is no longer allowed. Please use a newer client or a different key type.

Please see https://github.blog/2021-09-01-improving-git-protocol-security-github for more information.

### PR DESCRIPTION
**Pull request by kb-1000**
I unfortunately haven't been able to get the new JSCH working without adding Bouncy Castle, though.

Tested on an Android 5.0 AVD:
Screenshot_20220319_113701

Fixes #601, fixes #600

### CODE REVIEW

File: app/src/main/java/com/manichord/mgit/ssh/PrivateKeyGenerate.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        val newKey = File(PrivateKeyUtils.getPrivateKeyFolder(), newFilename)
+        val newPubKey = File(PrivateKeyUtils.getPublicKeyFolder(), newFilename)
+        try {
+            val jsch = JSch()
+            val kpair = KeyPair.genKeyPair(jsch, type, keySize)
+            kpair.writePrivateKey(FileOutputStream(newKey))
+            kpair.writePublicKey(FileOutputStream(newPubKey), "mgit")
+            kpair.dispose()
+        } catch (e: Exception) {
+            //TODO
```
Comment (maks) [inline app/src/main/java/com/manichord/mgit/ssh/PrivateKeyGenerate.kt:78]: I think we can't just swallow the exception here. 
Perhaps just show a generic error message or throw an Exception that will then allow the user to submit a crash report via email? (now that Sentry crash reporting has been removed).

**Review by maks — CHANGES_REQUESTED** [review]
@kb-1000  Thank you for contributing this PR! 👍🏻 
Just a couple of small bits of feedback from me.

File: app/src/main/java/com/manichord/mgit/transport/MGitSSLSocketFactory.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 public class MGitSSLSocketFactory extends SSLSocketFactory {
 
     private SSLSocketFactory wrappedSSLSocketFactory;
-    public static String[] enabledProtocols = new String[] {"TLSv1.2", "TLSv1.1", "TLSv1"};
+    public static String[] enabledProtocols = new String[] {"TLSv1.3", "TLSv1.2", "TLSv1.1", "TLSv1"};
```
[the remark this answers is not in the document]
Comment (kb-1000) [reply app/src/main/java/com/manichord/mgit/transport/MGitSSLSocketFactory.java:15]: Do you mean TLS 1 and TLS 1.1?
Removing 1.2 seems a bit too extreme if you ask me, and it doesn't seem to be deprecated according to Wikipedia.

File: app/src/main/java/me/sheimi/sgit/MGitApplication.kt [hunk 3]
```diff
@@ -10,8 +10,11 @@ import org.acra.config.dialog
 import org.acra.config.mailSender
 import org.acra.data.StringFormat
 import org.acra.ktx.initAcra
+import org.bouncycastle.jce.provider.BouncyCastleProvider
```
[the remark this answers is not in the document]
Comment (kb-1000) [reply app/src/main/java/me/sheimi/sgit/MGitApplication.kt:13]: Mostly its size, the JAR on maven is about 5MB.

File: app/src/main/java/com/manichord/mgit/transport/MGitSSLSocketFactory.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (maks) [reply app/src/main/java/com/manichord/mgit/transport/MGitSSLSocketFactory.java]: yes sorry that was a typo, I meant 1 and **1.1**

File: app/src/main/java/me/sheimi/sgit/MGitApplication.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (maks) [reply app/src/main/java/me/sheimi/sgit/MGitApplication.kt:13]: yes that is a bit unfortunate but worth it to get access to more modern, secure crypto algorithms

**Review by maks — APPROVED** [review]
@kb-1000 thanks again, thats great, all good to go now! 👍🏻

**Pull request merged**